store/src/java/com/zimbra/cs/milter/MilterHandler.java

```diff
@@ -354,6 +355,26 @@ private void SMFIC_Rcpt() throws ServiceException {
             connection.send(new MilterPacket(SMFIR_TEMPFAIL));
             return;
         }
+        try { // ZBUG-2165: prevent forwarding to non-permitted distribution list
+            ZimbraLog.milter.trace("evaluating forwarding to non-permitted distribution list");
+            final Account account = prov.getAccountByName(rcpt);
+            if (account != null) {
+                ZimbraLog.milter.trace("account: %s", account.getName());
+                final String prefMailForwardingAddress = account.getPrefMailForwardingAddress();
+                ZimbraLog.milter.trace("prefMailForwardingAddress: %s", prefMailForwardingAddress);
+                if (prefMailForwardingAddress != null && prov.isDistributionList(prefMailForwardingAddress)) {
```

Comment:
The `zimbraPrefMailForwardingAddress` is a comma-separated list of email addresses. Let's make sure to test this with multiple email addresses, and DLs set in that property and address handling it.

https://github.com/Zimbra/zm-mailbox/blob/develop/store/conf/attrs/zimbra-attrs.xml#L1868

Reply:
Done